canonical-web-and-design/maas-ui
# Add MachineListTable to KVM summary page.

## Done

- Added `showActions` prop to MachineListTable (default=true) which determines whether checkboxes and in-table actions will show
- Added MachineListTable to KVM summary page (with showActions=false).
- I was going to convert some files to TS but it would have required converting the machines selector file to TS too. I'll tackle that as a separate ticket.

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to /MAAS/r/kvm and click on a KVM.
- Check that a machine list displays with a correctly filtered list of machines.
- Check that the machine list does not have any checkboxes, and you cannot perform any in-table actions.

## Fixes

Fixes canonical-web-and-design/MAAS-squad#1959

## Screenshot
![0 0 0 0_8400_MAAS_r_kvm_27 (1)](https://user-images.githubusercontent.com/25733845/87280960-176ac300-c536-11ea-8f45-00eeadb26130.png)
